packages/@uppy/dashboard/src/components/FileItem/FileInfo/index.js

```diff
@@ -19,10 +19,26 @@ const renderFileSource = (props) => (
   </div>
 )
 
-const renderFileName = (props) =>
-  <div class="uppy-DashboardItem-name">
-    {truncateString(props.file.meta.name, props.currentWidth > 606 ? 35 : 60)}
-  </div>
+const renderFileName = (props) => {
+  // Take up at most 2 lines on any screen
+  let maxNameLength
+  // For very small mobile screens
+  if (props.containerWidth < 352) {
+    maxNameLength = 35
+  // For regular mobile screens
+  } else if (props.containerWidth < 577) {
+    maxNameLength = 60
+  // For desktops
+  } else {
+    maxNameLength = 35
+  }
```

Comment:
This works 👍 but I wonder if those need to be in sync with:

```js
{ 'uppy-size--md': props.containerWidth > 576 },
{ 'uppy-size--lg': props.containerWidth > 700 },
{ 'uppy-size--xl': props.containerWidth > 900 },
```

🤔

Reply:
577px is not arbitrary, it's basically 576 (I agree though, I'll make it >= 576, so that's it's searchable).
352px is arbitrary, but I don't think there is a way around that.